<img src="https://upload.wikimedia.org/wikipedia/commons/9/97/Pont_des_Arts%2C_Paris.jpg">Photo by <a href="//commons.wikimedia.org/wiki/User:Benh" title="User:Benh">Benh Lieu Song</a> - <span class="int-own-work" lang="en">Own work</span>, <a href="https://creativecommons.org/licenses/by-sa/3.0" title="Creative Commons Attribution-Share Alike 3.0">CC BY-SA 3.0</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=10190129">Link</a>

<h1><b>Real estate prices in Paris:</b><br />
A multivariate study of the location criterion.</h1>
<p>Stephanie Blanchet<br />
March 7, 2019</p>

<h1>1. Introduction</h1>

<p>The fourth most expensive metropolis in the world [<a href="https://www.century21-hec-paris-15.com/actus/1745088062/selon-century21-que-peut-on-soffrir-avec-300000-euros-dans-le-monde/", title="See Century 21 study published on October, 2018",  target="_blank">1</a>], Paris has seen its real estate prices quadrupled in 20 years. According to the Paris Notary Chamber [<a href="https://basebien.com/PNSPublic/_medias/pdf/chiffre/pns_conjoncture.pdf" title="See the Paris Notary Chamber study published on January, 2019", target="_blank">2</a>], a new record should be reached in the spring of 2019 with an average of 9730 Euros/m2, an increase of around 7% in one year. If the permanent soaring benefits all boroughs, all are not equal. Average prices currently range from 7930 Euros/m2 in the "19th arrondissement" to 13150 Euros/m2 in the "6th". At the neighborhoods scale, the gaps are even greater, with an average range of 7460 Euros/m2 in "La Chapelle" (18th) to 17410 Euros/m2 in "Odéon" (6th), nearly 15 times the net minimum income.</p>

<p>Real estate specialists generally consider that the location of a property is the primary criterion in estimating its market value. Benefits such as the number of services and shops, local amenities, public transport or schools are regularly cited to explain that one area is valued more than another. But what about the City of Lights? Does that kind of rationality also drive the invisible hand of the Parisian market?</p>

<p>This is the question that this study proposes to answer, by measuring the link between several variables related to location and the average price of real estate by borough. It should allow to establish a grid of parameters and a cartography for the use of professionals such as real estate agencies and notaries, but also of any potential buyer wishing to understand what really weighs in the price of a good according to its situation in the French capital, especially when this price appears to be inappropriate with the intrinsic qualities of the property in question (e.g. high price for a narrow area, obsolete facilities, absence of an elevator, ...).</p>

<hr>

> _The problem is_:<br>
What can explain the differences in the price of real estate from one Parisian borough to another, all other things being equal?<br><br>
_Stakeholders are_:<br>
Real estate agencies, notaries, and buyers themselves.

<hr>

<h1>2. Methodology</h1>

<h2>2.1 Data selection</h2>

<p>The data were selected on criteria of reliability, transparency on their origin, date of last update, as well as on the presence of a division by borough of Paris, in order to gather various factors likely to explain the variability of the prices depending on the location.</p>

<p>Thus, official sources and open institutional or government databases have been favored:</p>
    
   <ul>
<li>the French national institute for statistics and economic studies (INSEE),</li>
<li>Paris Notary Chamber,</li>
<li>Paris City Hall,</li>
<li>the Regional Council of Ile-de-France (Greater Paris),</li>
<li>the French Ministry of Education</li>
    </ul>

<p>FourSquare API provided user generated content (UGC) about the most common venues in each borough.</p>
<br>
<p><b><i>Table 1. Summary table of data</i></b></p>
<hr>

<table>
    <tr>
        <th><p style="text-align:left">DATA CONTENT & USE</p></th>
        <th><p style="text-align:left">SOURCE</p></th>
        <th><p style="text-align:left">INFORMATION LINK</p></th>
        <th><p style="text-align:left">UPDATE</p></th>
    <tr>
    <tr>
        <td><p style="text-align:left"><b>1. Average real estate prices by borough</b><br>
            - the target variable</p></td>
        <td><p style="text-align:left">Paris Notary<br> Chamber</p></td>
        <td><p style="text-align:left"><a href="http://notairesdugrandparis.fr/fr/carte-des-prix#paris", target="_blank">http://notairesdugrandparis.fr/fr/carte-des-prix#paris</a></p></td>
        <td><p style="text-align:left">January, 2019</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>2. Geodata</b><br>
- get the latitude and longitude of each Parisian borough so to send queries to the FourSquare API and create a map<br>
- also get areas to calculate the density per square kilometer of each independent variable</p></td>
        <td><p style="text-align:left">Paris City Hall</p></td>
        <td><p style="text-align:left"><a href="https://opendata.paris.fr/explore/dataset/arrondissements/information/", target="_blank">https://opendata.paris.fr/explore/dataset/arrondissements/information/</a></p></td>
        <td><p style="text-align:left">-</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>3. Population</b><br> 
- calculate the population density per km2 of each borough</p></td>
        <td><p style="text-align:left">INSEE</p></td>
        <td><p style="text-align:left"><a href="https://www.insee.fr/fr/statistiques/fichier/3677781/dep75.pdf", target="_blank">https://www.insee.fr/fr/statistiques/fichier/3677781/dep75.pdf</a></p></td>
        <td><p style="text-align:left">January, 2019</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>4. Private / Public Service offering</b><br>
- list and count the different categories of services available in each borough</p></td>
        <td><p style="text-align:left">Regional Council<br> of Ile-de-France</p></td>
        <td><p style="text-align:left"><a href="https://www.data.gouv.fr/fr/datasets/les-services-aux-particuliers-par-commune-ou-arrondissement-base-permanente-des-equipements-idf/", target="_blank">https://www.data.gouv.fr/fr/datasets/les-services-aux-particuliers-par-commune-ou-arrondissement-base-permanente-des-equipements-idf/</a></p></td>
        <td><p style="text-align:left">March, 2016</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>5. Shop offering</b><br>         
- list and count the different categories of stores and shops available in each borough</p></td>
        <td><p style="text-align:left">Regional Council<br> of Ile-de-France</p></td>
        <td><p style="text-align:left"><a href="https://www.data.gouv.fr/fr/datasets/les-commerces-par-commune-ou-arrondissement-base-permanente-des-equipements-idf/", target="_blank">https://www.data.gouv.fr/fr/datasets/les-commerces-par-commune-ou-arrondissement-base-permanente-des-equipements-idf/</a></p></td>
        <td><p style="text-align:left">March, 2016</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>6. Popular venues</b><br>
- list and count the most common categories of venues in each borough</p></td>
        <td><p style="text-align:left">FourSquare API</p></td>
        <td><p style="text-align:left"><a href="https://fr.foursquare.com/", target="_blank">https://fr.foursquare.com/</a></p></td>
        <td><p style="text-align:left">March, 2019</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>7. Municipal facilities</b><br>
- list and count the different categories of collective facilities in each borough</p></td>
        <td><p style="text-align:left">Paris City Hall</p></td>
        <td><p style="text-align:left"><a href="https://opendata.paris.fr/explore/dataset/equipements_de_proximite/information/", target="_blank">https://opendata.paris.fr/explore/dataset/equipements_de_proximite/information/</a></p></td>
        <td><p style="text-align:left">May, 2013</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>+ Rail transport</b> (refers to ‘Municipal facilities’)<br>
- list and count the rail transports (metro, tramway, train) available in each borough</p></td>
        <td><p style="text-align:left">Regional Council<br> of Ile-de-France</p></td>
        <td><p style="text-align:left"><a href="https://data.iledefrance.fr/explore/dataset/gares-et-stations-du-reseau-ferre-dile-de-france-par-ligne/information/", target="_blank">https://data.iledefrance.fr/explore/dataset/gares-et-stations-du-reseau-ferre-dile-de-france-par-ligne/information/</a></p></td>
        <td><p style="text-align:left">November, 2018</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>8. Education offering</b><br><br>
        <b>a. Primary & secondary schools</b><br>
- list and count public and private primary and secondary schools available in each borough</p></td>
        <td><p style="text-align:left">French Ministry<br> of Education</p></td>
        <td><p style="text-align:left"><a href="https://data.education.gouv.fr/explore/dataset/fr-en-adresse-et-geolocalisation-etablissements-premier-et-second-degre/information/", target="_blank">https://data.education.gouv.fr/explore/dataset/fr-en-adresse-et-geolocalisation-etablissements-premier-et-second-degre/information/</a></p></td>
        <td><p style="text-align:left">February, 2019</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>b. Public higher education establishments</b><br>
- list and count public higher education establishments available in each borough</p></td>
        <td><p style="text-align:left">Regional Council of Ile-de-France</p></td>
        <td><p style="text-align:left"><a href="https://data.iledefrance.fr/explore/dataset/implantations-des-etablissements-denseignement-superieur-publics/information/", target="_blank">https://data.iledefrance.fr/explore/dataset/implantations-des-etablissements-denseignement-superieur-publics/information/</a></p></td>
        <td><p style="text-align:left">October, 2017</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>c. Private higher education establishments</b><br>
- list and count private higher education establishments available in each borough</p></td>
        <td><p style="text-align:left">Regional Council<br> of Ile-de-France</p></td>
        <td><p style="text-align:left"><a href="https://data.iledefrance.fr/explore/dataset/idf_universites/information/", target="_blank">https://data.iledefrance.fr/explore/dataset/idf_universites/information/</a></p></td>
        <td><p style="text-align:left">May, 2013</p></td>
    </tr>
    <tr>
        <td><p style="text-align:left"><b>9. Social housing funded by Paris city since 2001</b><br>
- list and count the different categories of social housing built or acquired in each borough</p></td>
        <td><p style="text-align:left">Paris City Hall</p></td>
        <td><p style="text-align:left"><a href="https://opendata.paris.fr/explore/dataset/logements-sociaux-finances-a-paris/information/", target="_blank">https://opendata.paris.fr/explore/dataset/logements-sociaux-finances-a-paris/information/</a></p></td>
        <td><p style="text-align:left">February, 2019</p></td>
    </tr>
</table>   

<h2 id="Data_preparation">2.2 Data collection and preparation</h2>

<h3>2.2.1 Paris real estate prices</h3>

In [12]:
# Create dataframe on the basis of information provided by Paris Notary Chamber: http://notairesdugrandparis.fr/fr/carte-des-prix#paris

paris_realestate = ({'Arr': ['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20'],
'Avg RE Prices':[11890,11200,11840,12890,11360,13150,12590,10910,9870,9160,9380,8740,8630,9570,9520,10280,9890,8720,7930,8040]})

paris_realestate = pd.DataFrame.from_dict(paris_realestate)

paris_realestate['Arr'] = paris_realestate['Arr'].astype(str)

paris_realestate.head()

Arr  Avg RE Prices
0   1          11890
1   2          11200
2   3          11840
3   4          12890
4   5          11360

In [13]:
# Save the dataframe

paris_realestate.to_csv('paris_realestate.csv')

<h3>2.2.2 Paris geodata</h3>

In [14]:
# Load data

import requests

url = 'https://www.data.gouv.fr/en/datasets/r/4765fe48-35fd-4536-b029-4727380ce23c'
df = requests.get(url).json()
df = df['features']

In [15]:
# Define the dataframe columns

column_names = ['Arr', 'Borough', 'Latitude', 'Longitude', 'Area'] 

# Instantiate the dataframe

paris_geodata = pd.DataFrame(columns=column_names)

for data in df:
    number = data['properties']['c_ar']
    borough = data['properties']['l_aroff']
    area = data['properties']['surface']
        
    paris_geodata_latlon = data['properties']['geom_x_y']
    paris_geodata_lat = paris_geodata_latlon[0]
    paris_geodata_lon = paris_geodata_latlon[1]
    
    paris_geodata = paris_geodata.append({'Borough': borough,
                          'Arr': number,
                          'Latitude': paris_geodata_lat,
                          'Longitude': paris_geodata_lon,
                          'Area': area}, ignore_index=True)

paris_geodata['Arr'] = paris_geodata['Arr'].astype(str)

In [16]:
paris_geodata.head()

Arr         Borough   Latitude  Longitude          Area
0   2          Bourse  48.868279   2.342803  9.911537e+05
1   3          Temple  48.862872   2.360001  1.170883e+06
2  12         Reuilly  48.834974   2.421325  1.631478e+07
3   1          Louvre  48.862563   2.336443  1.824613e+06
4   4  Hôtel-de-Ville  48.854341   2.357630  1.600586e+06

In [17]:
# Save a clean and structured dataframe

paris_geodata.to_csv('paris_geodata.csv')

In [18]:
# Create a specific dataframe in anticipation of density-based data normalization 

paris_area = paris_geodata[['Arr', 'Area']]
paris_area.to_csv('paris_area.csv')

<h3>2.2.3 Paris population</h3>

In [13]:
# Load data

import tabula
from tabula import read_pdf

df = read_pdf('dep75.pdf', pages=9)

In [14]:
# Create clean dataframe

df = df[['Unnamed: 2', 'Unnamed: 5']]
df.columns = ['Arr', 'Population']
df = df.drop([0,1,2,3,24,25])

df_t = df.T
df_t.at['Arr'] = ['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20']

paris_population = df_t.T
paris_population.Population = paris_population.Population.str.replace(' ', '')
paris_population.Population = paris_population.Population.astype(int)

In [15]:
paris_population.head()

Arr  Population
4   1       16388
5   2       20410
6   3       35469
7   4       27795
8   5       59947

In [16]:
# Save a clean and structured dataframe

paris_population.to_csv('paris_population.csv')

<h3>2.2.4 Paris service offering</h3>

In [31]:
# Load data and create dataframe

df = pd.read_csv('les-services-aux-particuliers-par-commune-ou-arrondissement-base-permanente-des-.csv')
servcat = 'Arr;police;tresorerie;gendarmerie;cour_d_appel;tribunal_de_grande_instance;tribunal_d_instance;conseil_de_prud_hommes;tribunal_de_commerce;agence_de_proximite;agence_de_services_specialises;agence_thematique;banque_caisse_d_epargne;pompes_funebres;bureau_de_poste;reparation_auto_et_de_materiel_agricole;controle_technique_automobile;location_auto_utilitaires_legers;ecole_de_conduite;macon;platrier_peintre;menuisier_charpentier_serrurier;plombier_couvreur_chauffagiste;electricien;entreprise_generale_du_batiment;coiffure;veterinaire;agence_de_travail_temporaire;restaurant;agence_immobiliere;blanchisserie_teinturerie;soins_de_beaute'.split(";")
df = df['Arr;police;tresorerie;gendarmerie;cour_d_appel;tribunal_de_grande_instance;tribunal_d_instance;conseil_de_prud_hommes;tribunal_de_commerce;agence_de_proximite;agence_de_services_specialises;agence_thematique;banque_caisse_d_epargne;pompes_funebres;bureau_de_poste;reparation_auto_et_de_materiel_agricole;controle_technique_automobile;location_auto_utilitaires_legers;ecole_de_conduite;macon;platrier_peintre;menuisier_charpentier_serrurier;plombier_couvreur_chauffagiste;electricien;entreprise_generale_du_batiment;coiffure;veterinaire;agence_de_travail_temporaire;restaurant;agence_immobiliere;blanchisserie_teinturerie;soins_de_beaute'].str.split(";", n=len(servcat), expand=True)
df.columns = servcat 

In [34]:
# Explore features

df.columns.tolist()

['Arr',
 'police',
 'tresorerie',
 'gendarmerie',
 'cour_d_appel',
 'tribunal_de_grande_instance',
 'tribunal_d_instance',
 'conseil_de_prud_hommes',
 'tribunal_de_commerce',
 'agence_de_proximite',
 'agence_de_services_specialises',
 'agence_thematique',
 'banque_caisse_d_epargne',
 'pompes_funebres',
 'bureau_de_poste',
 'reparation_auto_et_de_materiel_agricole',
 'controle_technique_automobile',
 'location_auto_utilitaires_legers',
 'ecole_de_conduite',
 'macon',
 'platrier_peintre',
 'menuisier_charpentier_serrurier',
 'plombier_couvreur_chauffagiste',
 'electricien',
 'entreprise_generale_du_batiment',
 'coiffure',
 'veterinaire',
 'agence_de_travail_temporaire',
 'restaurant',
 'agence_immobiliere',
 'blanchisserie_teinturerie',
 'soins_de_beaute',
 'Law enforcement agencies',
 'Courts',
 'Tax offices',
 'Employment agencies',
 'Banks',
 'Automotive services',
 'Building craft',
 'Personal care services',
 'Estate agencies',
 'Restaurants',
 'Post offices',
 'Veterinaries',
 'Lau

In [32]:
# Narrow and translate features

df = df.apply(pd.to_numeric)
df['Arr'] = df['Arr'].astype(str)

df['Law enforcement agencies'] = df.police + df.gendarmerie
df['Courts'] = df.cour_d_appel + df.tribunal_de_grande_instance + df.tribunal_d_instance + df.conseil_de_prud_hommes + df.tribunal_de_commerce
df['Tax offices'] = df.tresorerie
df['Employment agencies'] = df.agence_de_proximite + df.agence_de_travail_temporaire + df.agence_de_services_specialises + df.agence_thematique
df['Tax offices'] = df.tresorerie
df['Banks'] = df.banque_caisse_d_epargne
df['Automotive services'] = df.reparation_auto_et_de_materiel_agricole + df.controle_technique_automobile + df.location_auto_utilitaires_legers + df.ecole_de_conduite 
df['Building craft'] = df.macon + df.platrier_peintre + df.menuisier_charpentier_serrurier + df.plombier_couvreur_chauffagiste + df.electricien + df.entreprise_generale_du_batiment
df['Personal care services'] = df.soins_de_beaute + df.coiffure
df['Estate agencies'] = df.agence_immobiliere
df['Restaurants'] = df.restaurant
df['Post offices'] = df.bureau_de_poste
df['Veterinaries'] = df.veterinaire
df['Laundries'] = df.blanchisserie_teinturerie
df['Death care services'] = df.pompes_funebres

paris_service_offering = df.drop(['police','tresorerie','gendarmerie','cour_d_appel','tribunal_de_grande_instance','tribunal_d_instance','conseil_de_prud_hommes','tribunal_de_commerce','agence_de_proximite','agence_de_services_specialises','agence_thematique','banque_caisse_d_epargne','pompes_funebres','bureau_de_poste','reparation_auto_et_de_materiel_agricole','controle_technique_automobile','location_auto_utilitaires_legers','ecole_de_conduite','macon','platrier_peintre','menuisier_charpentier_serrurier','plombier_couvreur_chauffagiste','electricien','entreprise_generale_du_batiment','coiffure','veterinaire','agence_de_travail_temporaire','restaurant','agence_immobiliere','blanchisserie_teinturerie','soins_de_beaute'], axis=1)

paris_service_offering['Service offering'] = paris_service_offering.sum(axis=1)

paris_service_offering.head()

Arr  Law enforcement agencies  Courts  Tax offices  Employment agencies  \
0   1                         2       3            1                   19   
1   5                         2       1            1                    8   
2   6                         3       1            1                    1   
3  13                         3       1            3                   13   
4   9                         4       1            2                  134   

   Banks  Automotive services  Building craft  Personal care services  \
0     78                   33             215                     136   
1     54                   28             246                     121   
2     65                   19             149                     137   
3     90                   94             682                     248   
4    215                   54             846                     286   

   Estate agencies  Restaurants  Post offices  Veterinaries  Laundries  \
0              279          709             7             3         18   
1              178          929             4             8         31   
2              191          625             7             9         21   
3              176          763            11            12         64   
4              361         1262            11             6         51   

   Death care services  Service offering  
0                    0              1503  
1                    6              1617  
2                    0              1229  
3                   10              2170  
4                    5              3238

In [33]:
# Save a clean and structured dataframe

paris_service_offering.to_csv('paris_service_offering.csv')

<h3>2.2.6 Paris shop offering</h3>

In [36]:
# Load data and create dataframe

df = pd.read_csv('les-commerces-par-commune-ou-arrondissement-base-permanente-des-equipements.csv')
shopcat = 'Arr;hypermarche;supermarche;grande_surface_de_bricolage;superette;epicerie;boulangerie;boucherie_charcuterie;produits_surgeles;poissonnerie;librairie_papeterie_journaux;magasin_de_vetements;magasin_d_equipements_du_foyer;magasin_de_chaussures;magasin_d_electromenager_et_de_mat_audio_video;magasin_de_meubles;magasin_d_articles_de_sports_et_de_loisirs;magasin_de_revetements_murs_et_sols;droguerie_quincaillerie_bricolage;parfumerie;horlogerie_bijouterie;fleuriste;magasin_d_optique;station_service'.split(";")
df = df['Arr;hypermarche;supermarche;grande_surface_de_bricolage;superette;epicerie;boulangerie;boucherie_charcuterie;produits_surgeles;poissonnerie;librairie_papeterie_journaux;magasin_de_vetements;magasin_d_equipements_du_foyer;magasin_de_chaussures;magasin_d_electromenager_et_de_mat_audio_video;magasin_de_meubles;magasin_d_articles_de_sports_et_de_loisirs;magasin_de_revetements_murs_et_sols;droguerie_quincaillerie_bricolage;parfumerie;horlogerie_bijouterie;fleuriste;magasin_d_optique;station_service'].str.split(";", n=len(shopcat), expand=True)
df.columns = shopcat

In [37]:
# Explore features

df.columns.tolist()

['Arr',
 'hypermarche',
 'supermarche',
 'grande_surface_de_bricolage',
 'superette',
 'epicerie',
 'boulangerie',
 'boucherie_charcuterie',
 'produits_surgeles',
 'poissonnerie',
 'librairie_papeterie_journaux',
 'magasin_de_vetements',
 'magasin_d_equipements_du_foyer',
 'magasin_de_chaussures',
 'magasin_d_electromenager_et_de_mat_audio_video',
 'magasin_de_meubles',
 'magasin_d_articles_de_sports_et_de_loisirs',
 'magasin_de_revetements_murs_et_sols',
 'droguerie_quincaillerie_bricolage',
 'parfumerie',
 'horlogerie_bijouterie',
 'fleuriste',
 'magasin_d_optique',
 'station_service']

In [22]:
# Narrow and translate features

df = df.apply(pd.to_numeric)
df['Arr'] = df['Arr'].astype(str)

df['Supermarkets'] = df.supermarche + df.hypermarche
df['Convenience stores'] = df.superette + df.epicerie + df.produits_surgeles
df['Food shops'] = df.boulangerie + df.boucherie_charcuterie + df.poissonnerie
df['Personal goods'] = df.magasin_de_vetements + df.magasin_de_chaussures + df.magasin_d_articles_de_sports_et_de_loisirs + df.magasin_d_articles_de_sports_et_de_loisirs + df.parfumerie + df.horlogerie_bijouterie + df.magasin_d_optique
df['Household goods'] = df.magasin_d_equipements_du_foyer + df.magasin_d_electromenager_et_de_mat_audio_video + df.magasin_de_meubles + df.magasin_de_revetements_murs_et_sols + df.droguerie_quincaillerie_bricolage
df['Books & stationery'] = df.librairie_papeterie_journaux
df['Florists'] = df.fleuriste
df['Gas stations'] = df.station_service

paris_shop_offering = df.drop(['hypermarche','supermarche','grande_surface_de_bricolage','superette','epicerie','boulangerie','boucherie_charcuterie','produits_surgeles','poissonnerie','librairie_papeterie_journaux','magasin_de_vetements','magasin_d_equipements_du_foyer','magasin_de_chaussures','magasin_d_electromenager_et_de_mat_audio_video','magasin_de_meubles','magasin_d_articles_de_sports_et_de_loisirs','magasin_de_revetements_murs_et_sols','droguerie_quincaillerie_bricolage','parfumerie','horlogerie_bijouterie','fleuriste','magasin_d_optique','station_service'], axis=1)

paris_shop_offering['Shop offering'] = paris_shop_offering.sum(axis=1)

paris_shop_offering.head()

Arr  Supermarkets  Convenience stores  Food shops  Personal goods  \
0  17            31                 176         224             803   
1   2             4                  33          40             641   
2   3             5                  49          42             536   
3   6             7                  54          79             921   
4   8            23                  69          77            1151   

   Household goods  Books & stationery  Florists  Gas stations  Shop offering  
0              218                 145        77             4           1678  
1               57                  72        14             0            861  
2               94                  29        20             0            775  
3              141                 176        26             0           1404  
4              245                 127        39             4           1735

In [24]:
# Save a clean and structured dataframe

paris_shop_offering.to_csv('paris_shop_offering.csv')

<h3>2.2.7 Paris popular venues</h3>

In [ ]:
# Get the geospatial coordinates of Paris

from geopy.geocoders import Nominatim 

address = 'Paris, France'

geolocator = Nominatim(user_agent="paris_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude

print('The geograpical coordinates of Paris City are {}, {}.'.format(latitude, longitude))

In [ ]:
# FourSquare request

CLIENT_ID = 'secret'  
CLIENT_SECRET = 'secret'  
VERSION = '20180604'

search_query = ''
latitude = 48.8566101
longitude = 2.3514992
radius = 500  
limit = 200

In [ ]:
# Get popular venues by borough

def getNearbyVenues(names, latitudes, longitudes):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            limit)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Borough', 
                  'Latitude', 
                  'Longitude', 
                  'Venue', 
                  'Venue Lat.', 
                  'Venue Long.', 
                  'Venue Category']
    
    return(nearby_venues)

In [2]:
paris_borough_venues = getNearbyVenues(names=paris['Borough'], latitudes=paris['Latitude'], longitudes=paris['Longitude'])
paris_borough_venues.shape

(1411, 7)

In [2]:
import pandas as pd

paris_borough_venues = pd.read_csv('paris_borough_venues_4sq_250219.csv').drop(['Unnamed: 0'], axis=1)
paris_borough_venues.head()

Borough   Latitude  Longitude                 Venue  Venue Lat.  \
0  Bourse  48.868279   2.342803           Le Silencio   48.868998   
1  Bourse  48.868279   2.342803     Pizzeria Popolare   48.868252   
2  Bourse  48.868279   2.342803            Le Moderne   48.868856   
3  Bourse  48.868279   2.342803  L'Appartement Sézane   48.869591   
4  Bourse  48.868279   2.342803               Saturne   48.868394   

   Venue Long.     Venue Category  
0     2.343417          Nightclub  
1     2.343398        Pizza Place  
2     2.342142  French Restaurant  
3     2.345034           Boutique  
4     2.341825  French Restaurant

In [3]:
paris_borough_venues.rename(columns = {'Venue Category': 'Venue Subcat.'}, inplace=True)

In [4]:
# Correct shells causing duplicates

paris_borough_venues['Venue'].replace(
{'SUBWAY':'Subway',
 'Le Moulin de La Vierge' : 'Le Moulin de la Vierge',
 'Wild & The Moon':'Wild & the Moon',
 'Comme Au Vietnam':'Comme au Vietnam',
 'Rocher du Zoo de Vincennes' : 'Zoo de Vincennes',
 'Parc zoologique de Paris' : 'Zoo de Vincennes'}, inplace = True)

In [5]:
# Count and drop duplicates

paris_borough_venues.drop_duplicates(subset='Venue', keep='first', inplace=True)
print('Returned {} venues.'.format(len(paris_borough_venues['Venue'].unique())))
print('Returned {} subcategories.'.format(len(paris_borough_venues['Venue Subcat.'].unique())))

Returned 1352 venues.
Returned 195 subcategories.


In [15]:
# Explore features

paris_borough_venues['Venue Subcat.'].unique().tolist()

['Nightclub',
 'Pizza Place',
 'French Restaurant',
 'Boutique',
 'Lebanese Restaurant',
 'Italian Restaurant',
 'Historic Site',
 'Cupcake Shop',
 'Cocktail Bar',
 'Tapas Restaurant',
 'Wine Bar',
 'Plaza',
 'Japanese Restaurant',
 'Burger Joint',
 'Bistro',
 'Coffee Shop',
 'Perfume Shop',
 'Bar',
 'Hotel',
 'Bakery',
 'Wine Shop',
 'Peruvian Restaurant',
 'Beer Bar',
 'Indonesian Restaurant',
 'Donut Shop',
 'Chinese Restaurant',
 'Salad Place',
 'Pastry Shop',
 'Souvlaki Shop',
 'Thai Restaurant',
 'Concert Hall',
 'Indie Movie Theater',
 'Brasserie',
 'Antique Shop',
 'Indian Restaurant',
 'New American Restaurant',
 "Women's Store",
 'Sandwich Place',
 'Juice Bar',
 'Greek Restaurant',
 'Creperie',
 'Lounge',
 'Gym / Fitness Center',
 'American Restaurant',
 'Furniture / Home Store',
 'Fish & Chips Shop',
 'Asian Restaurant',
 'Ice Cream Shop',
 'Gourmet Shop',
 'Gluten-free Restaurant',
 'Theater',
 'Hotel Bar',
 'Café',
 'Shoe Store',
 'Park',
 'Farmers Market',
 'Museum',
 'Ch

Again, having a closer look at the features, many other duplicates were identified (e.g. 'Rocher du Zoo de Vincennes' and 'Parc zoologique de Paris' which refer to the same venue appear in two different categories).

In [7]:
# Clean again

paris_borough_venues['Venue Subcat.'].replace(
{'Gym':'Gym / Fitness Center',
 'Hostel' : 'Hotel',
 'Resort' : 'Hotel',
 'Pedestrian Plaza':'Plaza',
 'General College & University':'drop',
 'Insurance Office':'drop',
 'Taco Place':'drop',
 'Metro Station':'drop',
 'Laundromat':'drop',
 'Train Station':'drop',
 'Rental Car Location':'drop',
 'Bus Station':'drop'}, inplace = True)

In [8]:
# Check NaN

import numpy as np

paris_borough_venues.replace('drop', np.nan, inplace=True)
paris_borough_venues.replace('nan', np.nan, inplace=True)
notassigned = paris_borough_venues.isnull() 

print('Not assigned in Venue Subcategory:', notassigned['Venue Subcat.'].value_counts())

Not assigned in Venue Subcategory: False    1341
True       11
Name: Venue Subcat., dtype: int64


In [9]:
# Drop cells where a subcategory is not assigned

paris_borough_venues = paris_borough_venues.dropna(subset= ['Venue Subcat.'], how= 'any')

In [10]:
print('Returned {} subcategories.'.format(len(paris_borough_venues['Venue Subcat.'].unique())))

Returned 182 subcategories.


In [12]:
# Narrow and translate features

paris_venues = paris_borough_venues.copy()
conditions = [
    (paris_venues['Venue Subcat.'].str.contains('Hotel|Motel|Bed')),
    (paris_venues['Venue Subcat.'].str.contains('Bar|Pub|Café|Bistro|Lounge|Coffee|Tea|Brewery|Speakeasy|Gastropub')),
    (paris_venues['Venue Subcat.'].str.contains('Barbershop|Spa')),
    (paris_venues['Venue Subcat.'].str.contains('Theater|Museum|Art|Exhibit|Concert|Auditorium|Cultural|Multiplex|Stadium|Comedy')),
    (paris_venues['Venue Subcat.'].str.contains('Pizza|Burger|Salad|Sandwich|Noodle|Chips|Fast Food')),
    (paris_venues['Venue Subcat.'].str.contains('Nightclub')),
    (paris_venues['Venue Subcat.'].str.contains('Restaurant|Diner|Steakhouse|Bouchon|Creperie|Brasserie')),
    (paris_venues['Venue Subcat.'].str.contains('Shop|Store|Boutique|Market|market|store|Bakery|Mall|Bodega|Lab|Pharmacy|Butcher')),
    (paris_venues['Venue Subcat.'].str.contains('Site|Plaza|Park|Garden|Zoo|Spot|Church|Bridge|Canal|Trail|Roof|Fountain|Monument|Boat|Lake|Playground')),
    (paris_venues['Venue Subcat.'].str.contains('Yoga|Bike|Gym|Pool|Dojo|Court'))]
choices = ['Popular accommodation',
 'Popular bars, pubs & cafés',
 'Popular beauty care places',            
 'Popular culture & entertainment places',
 'Popular easy food places',
 'Popular nightclubs',
 'Popular restaurants',
 'Popular shops & markets',
 'Popular touristic sites',
 'Popular sport activity places']

paris_venues['Venue Cat.'] = np.select(conditions, choices, default='Others')

paris_venues.head()

Borough   Latitude  Longitude                 Venue  Venue Lat.  \
0  Bourse  48.868279   2.342803           Le Silencio   48.868998   
1  Bourse  48.868279   2.342803     Pizzeria Popolare   48.868252   
2  Bourse  48.868279   2.342803            Le Moderne   48.868856   
3  Bourse  48.868279   2.342803  L'Appartement Sézane   48.869591   
4  Bourse  48.868279   2.342803               Saturne   48.868394   

   Venue Long.      Venue Subcat.                Venue Cat.  
0     2.343417          Nightclub        Popular nightclubs  
1     2.343398        Pizza Place  Popular easy food places  
2     2.342142  French Restaurant       Popular restaurants  
3     2.345034           Boutique   Popular shops & markets  
4     2.341825  French Restaurant       Popular restaurants

In [16]:
# Insert borough names

paris_venues['Arr'] = paris_borough_venues['Borough']
paris_venues['Arr'].replace({
    'Louvre':'1',
    'Bourse':'2',
    'Temple':'3',
    'Hôtel-de-Ville':'4',
    'Panthéon':'5',
    'Luxembourg':'6',
    'Palais-Bourbon':'7',
    'Élysée':'8',
    'Opéra':'9',
    'Entrepôt':'10',
    'Popincourt':'11',
    'Reuilly':'12',
    'Gobelins':'13',
    'Observatoire':'14',
    'Vaugirard':'15',
    'Passy':'16',
    'Batignolles-Monceau':'17',
    'Buttes-Montmartre':'18',
    'Buttes-Chaumont':'19',
    'Ménilmontant':'20'}, inplace=True) 

In [18]:
df = pd.get_dummies(paris_venues[['Venue Cat.']], prefix="", prefix_sep="")
df['Arr'] = paris_venues['Arr'].astype(str) 
fixed_columns = [df.columns[-1]] + list(df.columns[:-1])
df = df[fixed_columns]

df = df.groupby('Arr').sum().reset_index()

df['Popular venues'] = df.sum(axis=1)

In [19]:
df.head()

Arr  Popular accommodation  Popular bars, pubs & cafés  \
0   1                      9                          17   
1  10                      3                          27   
2  11                      1                          14   
3  12                      0                           0   
4  13                      1                           3   

   Popular beauty care places  Popular culture & entertainment places  \
0                           0                                       9   
1                           1                                       1   
2                           0                                       3   
3                           0                                       0   
4                           0                                       0   

   Popular easy food places  Popular nightclubs  Popular restaurants  \
0                         2                   1                   23   
1                         6                   0                   40   
2                         3                   1                   27   
3                         0                   0                    0   
4                         2                   0                   39   

   Popular shops & markets  Popular sport activity places  \
0                       19                              0   
1                       11                              1   
2                        4                              1   
3                        0                              1   
4                        5                              0   

   Popular touristic sites  Popular venues  
0                       15              95  
1                        4              94  
2                        4              58  
3                        2               3  
4                        3              53

In [20]:
# Save a clean and structured dataframe

df.to_csv('paris_popular_venues.csv')

<h3>2.2.8 Paris municipal facilities</h3>

In [19]:
# Load data

df = pd.read_json('equipements_de_proximite.json')['fields']

In [ ]:
# Define the dataframe columns

column_names = ['Arr', 'Facility Type'] 

# Instantiate the dataframe

paris_facilities = pd.DataFrame(columns=column_names)

for data in df:
    arr = data['ap_cp']
    facility = data['type_d_equipement']
    
    paris_facilities = paris_facilities.append({'Arr': arr,'Facility Type': facility}, ignore_index=True)

In [21]:
# Explore features

paris_facilities['Facility Type'].unique().tolist()

['Ecole maternelle & annexes',
 'Ecole élémentaire & annexes',
 'Ecole polyvalente',
 "Mairie d'arrondissement",
 'Maison des associations, locaux associatifs',
 'Promenade ouverte, mail planté, jardin, square',
 'Talus planté',
 'Crèche collective',
 "Jardin d'enfants OPAC",
 'Crèche familiale',
 'Terrain de tennis',
 "Centre d'animation",
 'Bain-douche',
 'Salle polyvalente (multi-usages)',
 'Jardinet décoratif, plate-bande',
 'Halte-garderie',
 "Terrain d'éducation physique",
 'Terrain de boules',
 'Gymnase',
 'Médiathèque',
 'Conservatoire',
 'Bibliothèque',
 'Pigeonnier',
 'Salles de sports',
 'Espace jeune',
 "Annexe de Mairie d'arrondissement"]

In [ ]:
# Narrow and translate features

paris_municipal_facilities = paris_facilities.copy()
conditions = [
    (paris_municipal_facilities['Facility Type'].str.contains('Crèche|multi-accueil|enfants|garderie|maternel')),
    (paris_municipal_facilities['Facility Type'].str.contains('jeunes|jeune|Bibliothèque|Conservatoire|Médiathèque|polyvalente|animation|associations')),
    (paris_municipal_facilities['Facility Type'].str.contains('Terrain|Gymnase|sports|Piscine|escalade|sportifs|Stade|roller|Bassin|Aires|Bain')),
    (paris_municipal_facilities['Facility Type'].str.contains('Promenade|Talus|Jardinet|Pigeonnier|Terre|vert')),
    (paris_municipal_facilities['Facility Type'].str.contains('Ecole|Mairie'))] # 'Ecole' is dropped as a dupplicate with education dataset
choices = ['Collective daycare',
 'Community centres',
 'Local sport facilities',            
 'Squares & gardens',
 'drop']

paris_municipal_facilities['Facility Type'] = np.select(conditions, choices, default='Others')

In [ ]:
# Insert borough numbers

paris_municipal_facilities['Arr'].replace({'75001':'1','75002':'2','75003':'3','75004':'4','75005':'5','75006':'6','75007':'7','75008':'8','75009':'9','75010':'10','75011':'11','75012':'12','75013':'13','75014':'14','75015':'15','75016':'16','75017':'17','75018':'18','75019':'19','75020':'20'}, inplace=True)
paris_municipal_facilities['Arr'] = paris_municipal_facilities['Arr'].astype(str)

In [5]:
df = pd.get_dummies(paris_municipal_facilities[['Facility Type']], prefix="", prefix_sep="")
df['Arr'] = paris_municipal_facilities['Arr'].astype(str) 
fixed_columns = [df.columns[-1]] + list(df.columns[:-1])
df = df[fixed_columns]

df = df.groupby('Arr').sum().reset_index()

In [6]:
# Create rail stations dataframe

paris_rail_stations =({'Arr':['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20'],
'Rail Stations':[13,11,10,9,13,15,16,20,24,21,26,30,23,21,32,26,28,22,39,26]})
paris_rail_stations = pd.DataFrame.from_dict(paris_rail_stations)

paris_rail_stations['Arr'] = paris_rail_stations['Arr'].astype(str)

In [7]:
# Join data

paris_municipal_facilities = df.join(paris_rail_stations.set_index('Arr'), on='Arr')
paris_municipal_facilities['Municipal Facilities'] = paris_municipal_facilities.sum(axis=1)

In [9]:
paris_municipal_facilities.head()

Arr  Collective daycare  Community centres  Local sport facilities  \
0   1                   4                  3                       1   
1  10                  20                  9                      14   
2  11                  28                  8                      21   
3  12                  29                 14                      19   
4  13                  49                 16                      32   

   Squares & gardens  Rail stations  Municipal facilities  
0                  5             13                    26  
1                 20             21                    84  
2                 31             26                   114  
3                 29             30                   121  
4                 58             23                   178

In [11]:
# Save a clean and structured dataframe

paris_municipal_facilities.to_csv('paris_municipal_facilities.csv')

<h3>2.2.9 Paris education offering</h3>

In [ ]:
# Load data

df = pd.read_json('fr-en-adresse-et-geolocalisation-etablissements-premier-et-second-degre.json')['fields']
df[0]

In [ ]:
# Define the dataframe columns

column_names = ['Arr', 'Degree', 'Sector'] 

# Instantiate the dataframe

df = pd.DataFrame(columns=column_names)

for data in df:
    arr = data['code_postal_uai'] 
    degree = data['nature_uai'] 
    sector = data['secteur_public_prive_libe']
    
    df = df.append({'Arr':arr,'Degree':degree,'Sector':sector},ignore_index=True)

In [ ]:
# Narrow and translate features

df['Degree'].replace(
{'151':'Primary',
 '153':'Primary',
 '101':'Primary',
 '103':'Primary',
 '302':'Secondary',
 '340':'Secondary',
 '334':'Secondary',
 '300':'Secondary',
 '390':'Secondary',
 '320':'Secondary',
 '306':'Secondary',
 '301':'Secondary',
 '370':'Secondary',
 '312':'Secondary',
 '352':'Secondary',
 '111':'Primary',
 '315':'Secondary',
 '162':'Primary', 
}, inplace=True)

df['Sector'].replace({'Privé':'Private',}, inplace=True)

In [ ]:
df.to_csv('paris_prim_sec_education.csv')

In [47]:
# Load public higher education dataset

df1 = pd.read_json('implantations-des-etablissements-denseignement-superieur-publics.json')['fields']
df1[0]

In [ ]:
# Define the dataframe columns

column_names = ['Arr', 'Type'] 

# Instantiate the dataframe

df = pd.DataFrame(columns=column_names)

for data in df:
    arr = data['com_code'] 
    type_est = data['type_d_etablissement']
    
    df = df.append({'Arr':arr,'Type':type_est},ignore_index=True)

In [ ]:
df1.to_csv('paris_public_higher_education.csv')

In [ ]:
# Load private higher education dataset

df2 = pd.read_json('idf_universites.json')['fields']
df2[0]

In [ ]:
# Define the dataframe columns

column_names = ['Arr', 'Type'] 

# Instantiate the dataframe

df = pd.DataFrame(columns=column_names)

for data in df:
    arr = data['com_code'] 
    type_est = data['type_d_etablissement']
    
    df = df.append({'Arr':arr,'Type':type_est},ignore_index=True)

In [ ]:
df2.to_csv('paris_private_higher_education.csv')

In [ ]:
# Gather the 3 education offering datasets

df3 = [df, df1, df2]
paris_education_offering = reduce(lambda  left,right: pd.merge(left,right,on=['Arr'], how='outer'), df3)

In [ ]:
paris_education_offering.head()

In [53]:
df = pd.get_dummies(paris_education_offering[['Degree', 'Sector']], prefix="", prefix_sep="")
df['Arr'] = paris_education_offering['Arr'].astype(str) 
fixed_columns = [df.columns[-1]] + list(df.columns[:-1])
df = df[fixed_columns]

df = df.groupby('Arr').sum().reset_index()

df['Education Offering'] = df.Private + df.Public

df.head()

Arr  Higher  Primary  Secondary  Private  Public  Education Offering
0   1       1        6          3        2       8                  10
1  10       3       33         19       14      41                  55
2  11       0       54         27       26      55                  81
3  12       2       51         29       21      61                  82
4  13      46       74         44       32     132                 164

In [54]:
# Save a clean and structured dataframe

df.to_csv('paris_education_offering.csv')

<h3>2.2.10 Paris social housing</h3>

In [1]:
# Load data

paris_social_housing = pd.read_csv('logements-sociaux-finances-a-paris.csv')

In [2]:
socialcat = "Arrondissement;Nb logements finances;Mode;Programme".str.split(";")
df = paris_social_housing["Arrondissement;Nb logements finances;Mode;Programme"].str.split(";", n=len(socialcat), expand=True)
paris_social_housing.columns = socialcat 

In [3]:
# Explore features

paris_social_housing['Programme'].unique().tolist()

['foyer pour jeunes travailleurs ou résidence pour jeunes',
 'logement familial',
 'résidence sociale pour démuni',
 'résidence pour étudiants',
 'site de desserrement de foyer de travailleur migrant',
 'maison relais',
 "centre d'hébergement et de réinsertion sociale",
 "établissement d'hébergement pour personnes âgées dépendantes",
 'résidence pour personnes âgées',
 'foyer de travailleurs migrants',
 "centre d'hébergement d'urgence",
 'foyer pour personnes handicapées']

In [5]:
# Explore features

paris_social_housing['Procedure'].unique().tolist()

['construction neuve',
 'acquisition réhabilitation',
 'acquisition conventionnement']

In [ ]:
# Narrow and translate features

paris_social = paris_social_housing.copy()
conditions = [
    (paris_social['Programme'].str.contains('âgées|handicapées')),
    (paris_social['Programme'].str.contains('familial')),
    (paris_social['Programme'].str.contains('migrants|migrant')),
    (paris_social['Programme'].str.contains('relais|réinsertion|urgence|démuni')),
    (paris_social['Programme'].str.contains('jeunes|étudiants')),
    (paris_social['Procedure'].str.contains('construction')),
    (paris_social['Procedure'].str.contains('acquisition'))]  
choices = ['SH for dependent persons',
'SH for families',
'SH for migrants',
'SH for poor people',
'SH for young adults',
'SH via construction',
'SH via acquisition']

paris_social['Programme'] = np.select(conditions, choices, default='Others')
paris_social['Procedure'] = np.select(conditions, choices, default='Others')

In [ ]:
df = pd.get_dummies(paris_social[['Programme','Procedure']], prefix="", prefix_sep="")
df['Arr'] = paris_social['Arr'].astype(str) 
fixed_columns = [df.columns[-1]] + list(df.columns[:-1])
df = df[fixed_columns]

df = df.astype(int)

df = df.groupby('Arr').sum().reset_index()

df['Social housing'] = df['SH via construction'] + df['SH via acquisition']

In [7]:
df.head()

Arr  Social housing  SH for dependent persons  SH for families  \
0    1             774                         0              707   
1    2             499                         0              431   
2    3            1160                        18             1068   
3    4            1374                        45             1198   
4    5            1198                        16             1083   

   SH for migrants  SH for poor people  SH for young adults  \
0                0                  60                    7   
1                0                  68                    0   
2                0                  74                    0   
3                0                  29                  102   
4                0                  42                   57   

   SH via acquisition  SH via construction  
0                 470                  304  
1                 360                  139  
2                 980                  180  
3                 948                  426  
4                1003                  195

In [8]:
# Save a clean and structured dataframe

df.to_csv('paris_social_housing.csv')

<h3>2.2.11 Gather the complete dataset</h3>

In [1]:
import pandas as pd

df1 = pd.read_csv('paris_area.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
df2 = pd.read_csv('paris_population.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
df3 = pd.read_csv('paris_service_offering.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
df4 = pd.read_csv('paris_shop_offering.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
df5 = pd.read_csv('paris_popular_venues.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
df6 = pd.read_csv('paris_municipal_facilities.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
df7 = pd.read_csv('paris_education_offering.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
df8 = pd.read_csv('paris_social_housing.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')

df9 = [df2, df3, df4, df5, df6, df7, df8]

In [2]:
paris_data = df1.join(df9)

In [ ]:
# Rename Borough column in anticipation of joining with geojson file

paris_data.rename(columns={'Borough':'l_aroff'}, inplace=True)

In [ ]:
paris_data.to_csv('paris_multivar_data.csv')

In [ ]:
# Calculate density per km2 for all features

paris_data = paris_data.apply(lambda x: x/paris_data['Area'])*1000000
paris_data.drop(['Area'], axis=1, inplace=True)

In [5]:
# Insert real estate prices

df0 = pd.read_csv('paris_realestate.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
paris_data = df0.join(paris_data)

In [8]:
# Insert geodata

df10 = pd.read_csv('paris_geodata.csv').drop(['Unnamed: 0'], axis=1).set_index('Arr')
paris_data = paris_data.join(df10)

In [7]:
# Save the complete dataset

paris_data.to_csv('paris_data_norm.csv')

<h2>2.3 Variables and indicators</h2>

<p>In this study, <b>6 groups of variables</b> were analyzed as possible predictors of real estate prices in Paris.</p>

<p>After data preprocessing and transforming stages, the total <b>number of features was reduced from 323 to 49</b> so to avoid too much granularity and to make the results readable. There were then normalized on the basis of their density per square kilometer to allow the comparison between boroughs.</p>

<p>Details of feature engineering are available on the online <a href="#Data_preparation">notebook</a>.</p>

<h3>2.3.1 Private / public service offering</h3>

<p>This dataset was reduced from 31 to 14 attributes listed below:</p> 

<ul>
    <li>Law enforcement offices</li>
    <li>Courts</li>
    <li>Tax offices</li>
    <li>Employment agencies</li>
    <li>Banks</li>
    <li>Automotive services</li>
    <li>Building craft</li>
    <li>Personal care services</li>
    <li>Estate agencies</li>
    <li>Restaurants</li>
    <li>Post offices</li>
    <li>Veterinaries</li>
    <li>Laundry / Dry cleaning services</li>
    <li>Death care services</li>
</ul>

<h3>2.3.2 Shop offering</h3>

<p>This dataset was reduced from 23 to 8 attributes listed below:</p> 

<ul>
    <li>Supermarkets</li>
    <li>Convenience stores</li>
    <li>Food shops</li>
    <li>Personal goods shops</li>
    <li>Household goods shops</li>
    <li>Books and stationery shops</li>
    <li>Florists</li>
    <li>Gas stations</li>
</ul>

<h3>2.3.3 Popular Venues</h3>

<p>Due to many duplicates and redundant categories - which is inherent to user generated content, this dataset was reduced from 195 to 10 attributes listed below:</p> 

<ul>
<li>Popular accommodations</li>
<li>Popular bars, pubs & cafés</li>
<li>Popular beauty care places</li>            
<li>Popular culture & entertainment places</li>
<li>Popular easy food places</li>
<li>Popular nightclubs</li>
<li>Popular restaurants</li>
<li>Popular shops & markets</li>
<li>Popular touristic sites</li>
<li>Popular sport activity places</li>
</ul>

<h3>2.3.4 Municipal facilities</h3>

<p>This dataset was reduced from 41 to 4 attributes listed below:</p> 

<ul>
    <li>Collective daycare</li>
    <li>Community centres</li>
    <li>Local sport facilities</li>
    <li>Squares & gardens</li>
    <li>Rail stations</li>
</ul>

<h3>2.3.5 Education offering</h3>

<p>The primary and secondary education dataset was reduced from 18 to 2 categories that are "primary" and "secondary" and divided by status "private" vs "public" ; public and private higher education datasets were treated as the two modalities of the category "Higher education":</p> 

<ul>
    <li>Primary schools</li>
    <li>Secondary schools</li>
    <li>Higher education</li>
    <li>Public vs private</li>
</ul>

<h3>2.3.6 Social housing</h3>

<p>This dataset was reduced from 15 to 7 attributes listed below, "via construction" and "via acquisition" being the two methods of social housing implementation:</p> 

<ul>
    <li>for dependent persons</li>
    <li>for families</li>
    <li>for young adults</li>
    <li>for poor people</li>
    <li>for migrants</li>
    <li>via construction vs via acquisition</li>
</ul>